brut.apktool/apktool-lib/src/main/java/brut/androlib/ApkBuilder.java

```diff
@@ -78,48 +80,53 @@ public ApkBuilder(Config config, ExtFile apkDir) {
-            String minSdkVersion = mApkInfo.getSdkInfo().get("minSdkVersion");
-            mMinSdkVersion = mApkInfo.getMinSdkVersionFromAndroidCodename(minSdkVersion);
-        }
+            if (outFile == null) {
+                String outFileName = mApkInfo.apkFileName;
+                outFile = new File(mApkDir, "dist" + File.separator + (outFileName == null ? "out.apk" : outFileName));
+            }
 
-        if (outFile == null) {
-            String outFileName = mApkInfo.apkFileName;
-            outFile = new File(mApkDir, "dist" + File.separator + (outFileName == null ? "out.apk" : outFileName));
-        }
+            //noinspection ResultOfMethodCallIgnored
+            new File(mApkDir, APK_DIRNAME).mkdirs();
+            File manifest = new File(mApkDir, "AndroidManifest.xml");
+            File manifestOriginal = new File(mApkDir, "AndroidManifest.xml.orig");
+
+            scheduleBuildSources();                         //create classes.dex
+            scheduleBuildNonDefaultSources();               //create all other dex files
+            buildManifestFile(manifest, manifestOriginal);  //backup AndroidManifest.xml
+            buildResources();                               //create res folder, manifest file and resources.arsc
+            buildLibs();                                    //copy lib, libs, kotlin and META-INF/services
+            buildCopyOriginalFiles();                       //copy some original files if they don't exist
+            mWorker.waitForFinish();                        //wait until all previous build tasks are finished
+            if (mBuildError.get() != null) {
+                throw mBuildError.get();
+            }
 
-        //noinspection ResultOfMethodCallIgnored
-        new File(mApkDir, APK_DIRNAME).mkdirs();
-        File manifest = new File(mApkDir, "AndroidManifest.xml");
-        File manifestOriginal = new File(mApkDir, "AndroidManifest.xml.orig");
-
-        buildSources();
-        buildNonDefaultSources();
-        buildManifestFile(manifest, manifestOriginal);
-        buildResources();
-        buildLibs();
-        buildCopyOriginalFiles();
-        buildApk(outFile);
-
-        // we must go after the Apk is built, and copy the files in via Zip
-        // this is because Aapt won't add files it doesn't know (ex unknown files)
-        buildUnknownFiles(outFile);
-
-        // we copied the AndroidManifest.xml to AndroidManifest.xml.orig so we can edit it
-        // lets restore the unedited one, to not change the original
-        if (manifest.isFile() && manifest.exists() && manifestOriginal.isFile()) {
-            try {
-                if (new File(mApkDir, "AndroidManifest.xml").delete()) {
-                    FileUtils.moveFile(manifestOriginal, manifest);
+            buildApk(outFile);                              //finish the apk by adding common and unknown files
```

Comment:
For all these comments can we either remove them and adapt methods to sustain what the comment would need? and/or just add a leading space so its like:

```
// create classes.dex
```